src/_pytest/python.py

```diff
@@ -929,6 +929,130 @@ def hasnew(obj: object) -> bool:
     return False
 
 
+@final
+@attr.s(frozen=True, auto_attribs=True, slots=True)
+class IdMaker:
+    argnames: Sequence[str]
+    parametersets: Sequence[ParameterSet]
+    idfn: Optional[Callable[[Any], Optional[object]]]
+    ids: Optional[Sequence[Union[None, str]]]
+    config: Optional[Config]
+    nodeid: Optional[str]
+
+    def make_unique_parameterset_ids(self) -> List[str]:
+        """Make a string for each ParameterSet to identify test invocations in pytest logs.
```

Comment:
The nodeid (where the parameterset ID ends up) is somewhat more than just for the pytest logs. So maybe something like this:

Make a unique identifier for each ParameterSet, that may be used to identify
the parametrization in a node ID.